src/realm/table.hpp

```diff
@@ -242,10 +244,11 @@ class Table {
     // Create an object with primary key. If an object with the given primary key already exists, it
     // will be returned and did_create (if supplied) will be set to false.
     // Potential tombstone will be resurrected
-    Obj create_object_with_primary_key(const Mixed& primary_key, FieldValues&&, bool* did_create = nullptr);
+    Obj create_object_with_primary_key(const Mixed& primary_key, FieldValues&&, UpdateMode mode = UpdateMode::all,
+                                       bool* did_create = nullptr);
```

Comment:
Why can't we just do `bool& did_create = false`?

Reply:
The `did_create` is optional. It can be nullptr.

Reply:
Oh yes, my bad sorry, we could have done `optional<boo&>` but it is too much fuss for nothing...